gemato/hash.py

```diff
@@ -34,14 +34,17 @@ def get_hash_by_name(name):
 	Get a hashlib-compatible hash object for hash named @name. Supports
 	multiple backends.
 	"""
+	# special case hashes
+	if name == '__size__':
+		return SizeHash()
+
+	# general hash support
 	try:
-		return hashlib.new(name)
+		if name in hashlib.algorithms_available:
+			return hashlib.new(name)
 	except ValueError:
```

Comment:
Since you removed the case for `ValueError`, remove the handler as well.

Reply:
Done